src/framework/preferences/js/SyllabificationEnactor.js

```diff
@@ -235,66 +242,78 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
-     * to create one and assign the related promise to the `hyphenators` member for future retrieval.
+     * Assembles an Object containing the information for locating the pattern file. If a pattern for the specific
+     * requested language code cannot be located, it will attempt to locate a fall back, by looking for a pattern
+     * supporting the generic language code. If no pattern can be found, `undefined` is returned as the `src` value.
      *
-     * When creating a hyphenator, it first checks if there is configuration for the specified `lang`. If that fails,
-     * it attempts to fall back to a less specific localization.
      *
-     * @param {Component} that - an instance of `fluid.prefs.enactor.syllabification`
      * @param {String} lang - a valid BCP 47 language code. (NOTE: supported lang codes are defined in the
      *                        `patterns`) option.
+     * @param {String} patterns - an object mapping language codes to file paths for the pattern files
      *
-     * @return {Promise} - returns a promise. If a hyphenator is successfully created, it is resolved with it.
-     *                     Otherwise, it resolves with undefined.
+     * @return {Object} - returns on Object containing the {"lang": "resolvedLanguageCode", src: "pattern/file/path"}
      */
-    fluid.prefs.enactor.syllabification.getHyphenator = function (that, lang) {
-        var promise = fluid.promise();
-        if (!lang) {
-            promise.resolve();
-            return promise;
-        }
-        lang = lang.toLowerCase();
+    fluid.prefs.enactor.syllabification.getPattern = function (lang, patterns) {
+        var src = patterns[lang];
 
-        // Use an existing hyphenator if available
-        var existing = that.hyphenators[lang];
-        if (existing) {
-            return existing;
+        if (!src) {
+            lang = lang.split("-")[0];
+            src = patterns[lang];
         }
 
-        // Attempt to create an appropriate hyphenator
+        return {
+            lang: lang,
+            src: src
+        };
+    };
+
+    // /**
+    //  * Retrieves a promise for the appropriate hyphenator. If a hyphenator has not already been created, it will attempt
+    //  * to create one and assign the related promise to the `hyphenators` member for future retrieval.
+    //  *
+    //  * When creating a hyphenator, it first checks if there is configuration for the specified `lang`. If that fails,
+    //  * it attempts to fall back to a less specific localization.
+    //  *
+    //  * @param {Component} that - an instance of `fluid.prefs.enactor.syllabification`
+    //  * @param {String} lang - a valid BCP 47 language code. (NOTE: supported lang codes are defined in the
+    //  *                        `patterns`) option.
+    //  *
+    //  * @return {Promise} - returns a promise. If a hyphenator is successfully created, it is resolved with it.
```

Comment:
It seems more appropriate to reject rather than resolve with undefined if the hyphenator can't be created

Reply:
This is actually related to a conversation I had with you when the syllabification code was first being implemented. Originally this was rejected, but it was suggested to resolve with an empty value instead. I believe for the use of promise sequences to trigger the afterSyllabification event.

see: http://irc-logs.fluidproject.org/%23fluid-work/%23fluid-work.2018-11-02.log

Reply:
@jobara Thanks for locating this ancient conversation. It seems that the main reason for the choice was the unsuitability of our fluid.promise.sequence algorithm for this use case and several others. It will be ok if you leave an implementation comment here explaining that a promise rejection would be more appropriate here and that we can improve the implementation once https://issues.fluidproject.org/browse/FLUID-5938 is resolved